test/testAdaptors/CurveAdaptor.t.sol

```diff
@@ -0,0 +1,1608 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { WstEthExtension } from "src/modules/price-router/Extensions/Lido/WstEthExtension.sol";
+import { CellarWithOracle } from "src/base/permutations/CellarWithOracle.sol";
+import { Cellar } from "src/base/Cellar.sol";
+import { CurveEMAExtension } from "src/modules/price-router/Extensions/Curve/CurveEMAExtension.sol";
+import { CurveAdaptor, CurvePool, CurveGauge } from "src/modules/adaptors/Curve/CurveAdaptor.sol";
+import { Curve2PoolExtension } from "src/modules/price-router/Extensions/Curve/Curve2PoolExtension.sol";
+import { MockDataFeed } from "src/mocks/MockDataFeed.sol";
+
+// Import Everything from Starter file.
+import "test/resources/MainnetStarter.t.sol";
+
+import { AdaptorHelperFunctions } from "test/resources/AdaptorHelperFunctions.sol";
+
+contract CurveAdaptorTest is MainnetStarterTest, AdaptorHelperFunctions {
```

Comment:
TO EIN - I've skimmed this but I will go through this more thoroughly tomorrow morning.